packages/core/test/util/initializedTooLateHeuristic_test.js

```diff
@@ -0,0 +1,78 @@
+/*
+ * (c) Copyright IBM Corp. 2022
+ */
+
+'use strict';
+
+const expect = require('chai').expect;
+const sinon = require('sinon');
+const path = require('path');
+const requireHook = require('../../src/util/requireHook');
+const initializedTooLateHeurstic = require('../../src/util/initializedTooLateHeuristic');
+
+describe('[UNIT] util.initializedTooLateHeurstic', () => {
+  const instrumentedModules = [];
+
+  before(() => {
+    sinon.stub(requireHook, 'onFileLoad').callsFake(function fake(m) {
+      instrumentedModules.push(m);
+    });
+    sinon.stub(requireHook, 'onModuleLoad').callsFake(function fake(m) {
+      instrumentedModules.push(m);
+    });
+
+    require.cache['/@contrast/agent/'] = {};
+
+    const tracing = require('../../src/');
+    tracing.init();
+  });
+
+  after(() => {
+    delete require.cache['/@contrast/agent/'];
+  });
+
+  beforeEach(() => {
+    initializedTooLateHeurstic.reset();
+  });
+
+  it('hasBeenInitializedTooLate is false', () => {
+    expect(initializedTooLateHeurstic()).to.be.false;
+  });
+
+  it('hasBeenInitializedTooLate is false', () => {
+    const p = '/Users/myuser/dev/instana/nodejs/node_modules/nope';
+    require.cache[p] = {};
+    expect(initializedTooLateHeurstic()).to.be.false;
+    delete require.cache[p];
+  });
+
+  it('hasBeenInitializedTooLate is true', () => {
+    const originalRequireCache = require.cache;
+
+    instrumentedModules.forEach(moduleName => {
+      // eslint-disable-next-line no-console
+      console.log(`for module ${moduleName}`);
+
+      // RESET CACHE & module
+      require.cache = originalRequireCache;
```

Comment:
_question:_ Does that actually do anything? Aren't `require.cache` and `originalRequireCache` two references to the very same object? AFAIK `require.resolve` does not create a new object for `require.cache`, it just updates the existing data structure right? (Also, our own modifications will not replace require.cache with a new object, as we only set entries and delete them.)

Reply:
Check if we need to use clone